test/ejabberd_sm_SUITE.erl

```diff
@@ -181,6 +185,27 @@ session_info_is_updated_if_keys_match(C) ->
     [#session{sid = Sid, info = #{key1 := val2}}]
      = ?B(C):get_sessions(U,S).
 
+%% Same resource but different sids
+session_info_is_updated_properly_if_session_conflicts(C) ->
+    %% We use 2 seeds here
+    {Sid, {U, S, _} = USR} = generate_random_user(<<"localhost">>),
+    %% Sid2 > Sid in this case, because SIDs have a timestamp in them
+    %% We cannot test store_info for Sid2 though, because it has a different pid
+    Sid2 = make_sid_with_unique_pid(),
+
+    %% Two sessions for the same USR are registered after that:
+    given_session_opened(Sid, USR, 1, [{key1, val1}, {key2, a}]),
+    given_session_opened(Sid2, USR, 1, [{key1, val2}, {key3, b}]),
+
+    %% Overwritten without merging
+    %% Because we don't call open_session twice for the same Sid, so no need to merge info
```

Comment:
This comment is hard to understand, because we do call it for `Sid` twice: in lines 197 and 202.